src/Core/Domain/Product/Image/DTO/ImageCollection.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\DTO;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\DeletedImage;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\ExistingConfigurableImage;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\NewConfigurableImage;
+
+/**
+ * Holds all product image state cases.
+ */
+class ImageCollection
+{
+    /** @var NewConfigurableImage[] */
+    private $newImages;
+
+    /** @var ExistingConfigurableImage[] */
+    private $existingImages;
+
+    /** @var DeletedImage[] */
+    private $deletesImages;
```

Comment:
How do you track deleted images? What is considered a deleted image? I think that user is able to delete image one by one.

Reply:
since we avoiding "ghost" products , then in create action we will create a bunch of product images which will not be submitted instantly with an ajax due to we not yet have assigned product. But you consider that in the update action we will instantly start updating and deleting product images?

Reply:
> But you consider that in the update action we will instantly start updating and deleting product images?

When you edit product, you just click delete and it should perform delete imo, you shouldn't need to click "Save product", since there is separate button on image for deleting. I guess same for upload when editing.

Reply:
but remember when we have a new product we dont have an id of it. So you suggest when we upload new image for non existing product we will have to be forced to enter required product name before since we need an id_product?